app/utils/store.py

```diff
@@ -17,22 +18,34 @@ def __init__(self, bucket=None):
     def init_app(self, app):
         self.bucket = app.config['DOCUMENTS_BUCKET']
 
-    def put(self, service_id, document_stream, *, mimetype):
+    def put(self, service_id, document_stream, *, mimetype, verification_email=None):
         """
         returns dict {'id': 'some-uuid', 'encryption_key': b'32 byte encryption key'}
         """
 
         encryption_key = self.generate_encryption_key()
         document_id = str(uuid.uuid4())
 
-        self.s3.put_object(
-            Bucket=self.bucket,
-            Key=self.get_document_key(service_id, document_id),
-            Body=document_stream,
-            ContentType=mimetype,
-            SSECustomerKey=encryption_key,
-            SSECustomerAlgorithm='AES256'
-        )
+        if verification_email:
+            hashed_recipient_email = self._hash_recipient_email(verification_email)
+            self.s3.put_object(
+                Bucket=self.bucket,
+                Key=self.get_document_key(service_id, document_id),
+                Body=document_stream,
+                ContentType=mimetype,
+                SSECustomerKey=encryption_key,
+                SSECustomerAlgorithm='AES256',
+                Metadata={"hashed-recipient-email": hashed_recipient_email}
+            )
+        else:
+            self.s3.put_object(
+                Bucket=self.bucket,
+                Key=self.get_document_key(service_id, document_id),
+                Body=document_stream,
+                ContentType=mimetype,
+                SSECustomerKey=encryption_key,
+                SSECustomerAlgorithm='AES256'
+            )
```

Comment:
This is a very repetitive way of excluding the Metadata argument. We could simplify this slightly :)

```
        extra_kwargs = {}
        if verification_email:
            extra_kwargs['Metadata'] = {'hashed-recipient-email': self._hash_recipient_email(verification_email)}
        
        self.s3.put_object(
            Bucket=self.bucket,
            Key=self.get_document_key(service_id, document_id),
            Body=document_stream,
            ContentType=mimetype,
            SSECustomerKey=encryption_key,
            SSECustomerAlgorithm='AES256',
            **extra_kwargs
        )
```